fe2-android/app/src/test/java/com/github/dedis/popstellar/model/objects/security/PrivateKeyTest.java

```diff
@@ -0,0 +1,48 @@
+package com.github.dedis.popstellar.model.objects.security;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.github.dedis.popstellar.model.objects.security.privatekey.PlainPrivateKey;
+import com.github.dedis.popstellar.model.objects.security.privatekey.ProtectedPrivateKey;
+import com.google.crypto.tink.KeysetHandle;
+import com.google.crypto.tink.PublicKeySign;
+import com.google.crypto.tink.subtle.Ed25519Sign;
+
+import net.i2p.crypto.eddsa.Utils;
+
+import org.junit.Test;
+
+import java.security.GeneralSecurityException;
+
+public class PrivateKeyTest {
+
+  private static final Base64URLData DATA = new Base64URLData("REFUQQ==");
+  private static final byte[] VALID_PRIVATE_KEY =
+      Utils.hexToBytes("3b28b4ab2fe355a13d7b24f90816ff0676f7978bf462fc84f1d5d948b119ec66");
+
+  @Test
+  public void signGivesSameValueForBothKeyType() throws GeneralSecurityException {
+    PrivateKey key1 = new PlainPrivateKey(VALID_PRIVATE_KEY);
+
+    KeysetHandle keyset = mock(KeysetHandle.class);
+    when(keyset.getPrimitive(PublicKeySign.class)).thenReturn(new Ed25519Sign(VALID_PRIVATE_KEY));
+    PrivateKey key2 = new ProtectedPrivateKey(keyset);
+
+    Signature sign1 = key1.sign(DATA);
+    Signature sign2 = key2.sign(DATA);
```

Comment:
aren't you effectively comparing `Ed25519Sign` with itself ? 🤔

Reply:
What do you mean, this is a very dumb test that make sure to different PrivateKey with the same data generate the same Signature.

I'm not sure I understand what you mean.

Reply:
I meant that effectively you've mocked the dependencies of `ProtectedPrivateKey` to the point where it behaves exactly like `PlainPrivateKey` and you're really just testing that `Ed25519Sign` behaves in the same way in two different places.
Thinking about it, though, it's at the very least a good non-regression test 👍

Reply:
In the best case scenario, I would want to give a real KeyHandler. But it is not generated in the Unit tests, so I mocked its behavior.

In my mind, this is a very simple test that makes sure the two types provide the same effective functionality.